pinot-core/src/main/java/org/apache/pinot/core/plan/AggregationPlanNode.java

```diff
@@ -118,6 +120,38 @@ public Operator<AggregationResultsBlock> buildNonFilteredAggOperator() {
     return new AggregationOperator(_queryContext, aggregationInfo, numTotalDocs);
   }
 
+  /**
+   * Returns {@code true} if any of the aggregation functions have null values, {@code false} otherwise.
+   *
+   * The current implementation is pessimistic and returns {@code true} if any of the arguments to the aggregation
+   * functions is of function type. This is because we do not have a way to determine if the function will return null
+   * values without actually evaluating it.
+   */
+  private boolean hasNullValues(AggregationFunction[] aggregationFunctions) {
+    for (AggregationFunction aggregationFunction : aggregationFunctions) {
+      List<?> inputExpressions = aggregationFunction.getInputExpressions();
+      if (inputExpressions.isEmpty()) {
+        continue;
+      }
```

Comment:
What if aggregation function has more than one input expression ?

Reply:
Ups, there is a bug in the case it is a literal

Reply:
What I meant is - shouldn't the function be checking all input expressions instead of only the first one ? 

```java
inputExpressions.get(0);
```

Reply:
Oh! But that was on a different line :laughing:.

Formally, you are right. But then `isFitForNonScanBasedPlan` is doing the same thing, so it has no side effect. Anyway, in order to decouple both methods, it would be better to check all inputs. I'm changing that.